src/main/java/org/opensearch/knn/index/codec/KNN80Codec/KNN80DocValuesConsumer.java

```diff
@@ -184,6 +188,20 @@ private void createKNNIndexFromScratch(FieldInfo fieldInfo, KNNCodecUtil.Pair pa
 
         // Used to determine how many threads to use when indexing
         parameters.put(KNNConstants.INDEX_THREAD_QTY, KNNSettings.state().getSettingValue(KNNSettings.KNN_ALGO_PARAM_INDEX_THREAD_QTY));
+        Object name = parameters.get(NAME);
+        if (name != null && name.equals(METHOD_NSG)) {
+            if(parameters.containsKey(PARAMETERS)) {
+                parameters.remove(PARAMETERS);
+            }
+            /** TODO:
+             * when numIds is too small, NSG graph would core/throw exception
+             * because There are too much invalid entries in the knn graph.
+             */
+            if(pair.docs.length < 256) {
+                parameters.put(INDEX_DESCRIPTION_PARAMETER, FAISS_FLAT_DESCRIPTION);
+            }
```

Comment:
can you please add a comment what we are trying to do over here?

Reply:
As i keep the comment code is a testing for this scenario: https://github.com/opensearch-project/k-NN/blob/639f90c3273bf6f26731fe67b99a0f6a9dc251a5/src/test/java/org/opensearch/knn/index/FaissIT.java#L195-L202

When NSG graph n is too small , it will receive SIGFPE signal as following picture shows
![img_v2_70b65047-33a9-4772-9f20-d51845da4e8g](https://github.com/opensearch-project/k-NN/assets/12760367/2093b1c1-cc0a-4f90-85b2-ac1fc1bcb3ec)

this because default `build_type=1` build with NNDescent and when call  `NNDescent#gen_random` and
`addr[i] = rng() % (N - size);` N - size would be zero when N is small in the code:
https://github.com/facebookresearch/faiss/blob/3219e3d12e6fc36dfdfe17d4cf238ef70bf89568/faiss/impl/NNDescent.cpp#L90-L93

Even if we use `build_type = 0` with brute force search
https://github.com/facebookresearch/faiss/blob/3219e3d12e6fc36dfdfe17d4cf238ef70bf89568/faiss/IndexNSG.cpp#L157-L163

it will throw exception in 
`    check_knn_graph(knng.data(), n, GK);` that is 
```
    FAISS_THROW_IF_NOT_MSG(
            total_count < n / 10,
```
https://github.com/facebookresearch/faiss/blob/3219e3d12e6fc36dfdfe17d4cf238ef70bf89568/faiss/IndexNSG.cpp#L238

i think NSG need some entry points for search, and in small dataset and positive GK, many invalid graph entries generate. 
so i try to solve this scenario, i use a FLAT map for small dataset for avoid throw exception when index only 1~256 documents.